packages/mdl-button/_variables.scss

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+:root {
+  --mdl-primary-color: rgb(63, 81, 181);
```

Comment:
> What is the point of making each color into a sass var?

https://github.com/google/material-design-lite/issues/4456#issuecomment-229726032

Sass variables are used to store the hard-coded color values. CSS Variables leverage those sass variables to achieve equivalent values. When we define a themable property, we define it both using the sass variable as well as the css custom variable.

When the code is output, the css variable - if supported - will override the hard-coded value produced by the sass variable. Otherwise, the sass variable value will be used. 

Using this system, clients can theme either in sass, or using custom properties. This is also the reason we may not want to rewrite custom properties to their hard-coded values in our code.

> What do you mean by primary/secondary/warn buttons?

The [buttons spec](https://material.google.com/components/buttons.html) makes no mention of semantic types of button. That is up to the applications to decide. This means there should be no modifier classes like `--primary`, `--accent`, etc.